### ISSUE TITLE
Create getAbsoluteURL helper function

### ISSUE THREAD

**Issue by mramato** [AnalyticalGraphicsInc/cesium#2673]
There are a few places in the code where we create absolute urls from relative paths.  The code generally looks something like the below.  We should probably create a private helper function and replace all usage of `resolve` with it.  I know it's at least used in `CzmlDataSource` and `KmlDataSource` and once #2659 comes in, `Model` as well.  May be a few other places as well.

```javascript
var docUri = new Uri(document.location.href);
var tmp = new Uri(url);
var absoluteUrl = tmp.resolve(docUri).toString();
```

**Comment by lilleyse** [comment]
The files that need to be updated are:

* `buildModuleUrl.js`
* `CzmlDataSource.js`
* `KmlDataSource.js`
* `Model.js`
* `CesiumTerrainProviderSpec.js`
* `isCrossOriginUrlSpec.js`
* `TaskProcessor.js`

For `3d-tiles`, the code is slightly different but can also use the helper function:

```javascript
var url = (new Uri(contentUrl).isAbsolute()) ? contentUrl : baseUrl + contentUrl;
```
In the files:
* `Cesium3DTile.js`
* `Instanced3DModel3DTileContentProvider.js`

**Comment by AmayAgrawal** [comment]
Can u please somewhat give a hint of about what type of private helper function function we need to build here.Can u please give any one example of how to do it??

**Comment by lilleyse** [comment]
It would be a helper function like `getFilenameFromUri`.

However @TomPed is working on this issue right now though, so you may want to look into another one.

**Comment by TomPed** [comment]
For the 3D tiles if I'm going to use the `getAbsoluteUri` I would not be using `isAbsolute`, is this correct?

**Comment by lilleyse** [comment]
Correct, `getAbsoluteUri` should handle that case.

**Comment by TomPed** [comment]
Why is it better to use `getAbsoluteUri` over `isAbsolute`?

**Comment by lilleyse** [comment]
Tile urls can either be local or absolute, which is why we did the check with `isAbsolute`. But really the code is here is the same as in the other areas, it's creating a absolute url from a base and relative. The `isAbsolute` check is basically implicit in the `Uri.resolve` function, so using `getAbsoluteUri` is the right thing to do here.

**Comment by pjcozzi** [comment]
Fixed in #3418.
